crates/jstz_api/src/file/blob.rs

```diff
@@ -0,0 +1,511 @@
+//! `jstz`'s implementation of JavaScript's `Blob` API.
+//!
+//! More information:
+//!  - [MDN documentation][mdn]
+//!  - [W3C `File` specification][spec]
+//!  - [WHATWG `Infra` specification][infra-spec]
+//!
+//! [mdn]: https://developer.mozilla.org/en-US/docs/Web/API/Blob
+//! [spec]: https://w3c.github.io/FileAPI/
+//! [infra-spec]: https://infra.spec.whatwg.org/
+
+use crate::idl::{ArrayBufferLike, JsBufferSource};
+use boa_engine::{
+    js_string,
+    object::{
+        builtins::{JsArray, JsArrayBuffer, JsPromise},
+        Object,
+    },
+    property::Attribute,
+    value::TryFromJs,
+    Context, JsArgs, JsError, JsNativeError, JsResult, JsString, JsValue, NativeFunction,
+};
+use boa_gc::{Finalize, GcRefMut, Trace};
+use jstz_core::{
+    accessor,
+    native::{register_global_class, Accessor, JsNativeObject, NativeClass},
+    value::IntoJs,
+};
+use std::cmp::{max, min};
+
+#[derive(Trace, Finalize, Clone)]
+pub struct Blob {
+    bytes: Vec<u8>,
+    size: u64,
+    type_: String,
+}
+
+// https://infra.spec.whatwg.org/#collect-a-sequence-of-code-points
+fn collect_code_points(mut position: &[u16]) -> (Vec<u16>, &[u16]) {
+    // 1. Let result be the empty string.
+    let mut result: Vec<u16> = vec![];
+    // 2. While position doesn’t point past the end of input and the code point at position within
+    //    input meets the condition condition:
+    while !position.is_empty() && (position[0] == 0x000A || position[0] == 0x000D) {
+        // 1. Append that code point to the end of result.
+        result.push(position[0]);
+        // 2. Advance position by 1.
+        position = &position[1..];
+    }
+    // 3. Return result.
+    (result, position)
+}
+
+// https://w3c.github.io/FileAPI/#convert-line-endings-to-native
+fn convert_line_endings_to_native(s: &[u16]) -> Vec<u16> {
+    // 1. Let native line ending be be the code point U+000A LF.
+    let native_line_ending: u16 = 0x000A;
+    // 3. Set result to the empty string.
+    let mut result: Vec<u16> = vec![];
+    // 4. Let position be a position variable for s, initially pointing at the start of s.
+    let position = s;
+    // 5. Let token be the result of collecting a sequence of code points that are not equal
+    //    to U+000A LF or U+000D CR from s given position.
+    let (mut token, mut position) = collect_code_points(position);
+    // 6. Append token to result.
+    result.append(&mut token);
+    // 7. While position is not past the end of s:
+    while !position.is_empty() {
+        // 1. If the code point at position within s equals U+000D CR:
+        if position[0] == 0x000D {
+            // 1. Append native line ending to result.
+            result.push(native_line_ending);
+            // 2. Advance position by 1.
+            position = &position[1..];
```

Comment:
I'm fairly certain we could utilise `std` here more (probably using iterators). But for now, this  works